crates/cargo-contract/src/cmd/build/mod.rs

```diff
@@ -668,20 +776,41 @@ pub(crate) fn execute(args: ExecuteArgs) -> Result<BuildResult> {
                 exec_cargo_dylint(&crate_metadata, verbosity)?;
             }
 
+            if let Some((x, y)) = counter {
```

Comment:
This should come before the linting check. Right now this is what the output looks like

```
❯ cargo contract check
 [1/2] Checking ink! linting rules
    Updating git repository `https://github.com/paritytech/ink/`
Checking with toolchain `nightly-2022-06-30-aarch64-apple-darwin`
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s

 Checking contract: foo [1/2]
 [2/2] Executing `cargo check`
 ```